packages/kbn-index-adapter/src/create_or_update_index.test.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { elasticsearchServiceMock, loggingSystemMock } from '@kbn/core/server/mocks';
+import { updateIndices, createIndex, createOrUpdateIndex } from './create_or_update_index';
+
+const logger = loggingSystemMock.createLogger();
+const esClient = elasticsearchServiceMock.createClusterClient().asInternalUser;
+
+esClient.indices.putMapping.mockResolvedValue({ acknowledged: true });
+esClient.indices.putSettings.mockResolvedValue({ acknowledged: true });
+
+const simulateIndexTemplateResponse = { template: { mappings: { is_managed: true } } };
+// @ts-expect-error test data type mismatch
```

Comment:
Let me know if there is a specific reason behind the shape you use here, because the following perfectly passes without errors:

```suggestion
const simulateIndexTemplateResponse = {
  template: {
    settings: {},
    aliases: {},
    mappings: {},
  },
};
```